libiocage/lib/DevfsRules.py

```diff
@@ -0,0 +1,367 @@
+import re
+import os.path
+
+import libiocage.lib.errors
+import libiocage.lib.helpers
+
+class DevfsRulesFilter:
+
+    def __init__(self, source, active_filters=[]):
+        self.source = source
+        self.active_filters = active_filters
+
+    def with_rule(self, rule):
+        return DevfsRulesFilter(filter(
+            lambda x: x.has_rule(rule),
+            self.source
+        ), self.active_filters + [rule])
+
+    def with_include(self, rule_name):
+        rule = f"add include ${rule_name}"
+        return self.with_rule(rule)
+
+    def create_ruleset(self, rule_name, rule_number):
+        """
+        Create a ruleset from the provided filters
+
+        Args:
+
+            rule_name (string):
+                Name of the rule that get's created
+
+            rule_number (int):
+                Number of the rule that get's created
+        """
+
+        ruleset = DevfsRuleset(ruleset_name, ruleset_number)
+        for line in self.active_filters:
+            ruleset.append(line)
+        return ruleset
+
+class DevfsRuleset(list):
+    """
+    Representation of a devfs ruleset in the devfs.rules file
+
+    DevfsRuleset instances behave like standard lists and can store strings
+    that  multiple lines (string)
+    """
+
+    PATTERN = r"^\[(?P<name>[a-z](?:[a-z0-9\-_]*[a-z0-9])?)=(?P<number>[0-9]+)\]\s*(?:#\s*(?P<comment>.*))?$"
```

Comment:
@araujobsd @skarekrow do you have any idea how to reasonably shorten this line?

@igalic FYI

Reply:
the `/x`, or verbose flag, might be better than shortening it, https://docs.python.org/3/library/re.html#re.VERBOSE

Reply:
looks good. will apply that!